libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/NativeLoginManager.kt

```diff
@@ -270,108 +266,194 @@ internal class NativeLoginManager(
+    override suspend fun startPasswordReset(
         username: String,
         reCaptchaToken: String,
-        otpVerificationMethod: OtpVerificationMethod
-    ): OtpRequestResult {
+    ): NativeLoginResult {
 
         // Validate parameters.
-        if (!isValidUsername(username.trim())) {
-            return OtpRequestResult(InvalidUsername)
-        }
-        val reCaptchaSiteKeyId = when {
-            reCaptchaSiteKeyId == null -> {
-                SalesforceSDKLogger.e(TAG, "A reCAPTCHA site key wasn't and must be provided when using enterprise reCAPATCHA.")
-                return OtpRequestResult(UnknownError)
+        val trimmedUsername = username.trim()
+        trimmedUsername.mapInvalidUsernameToResult()?.let { return it }
+        val reCaptchaParameterGenerationResult = generateReCaptchaParameters(reCaptchaToken)
+
+        // Generate the start password reset request body.
+        val startPasswordResetRequestBodyString = runCatching {
+            StartPasswordResetRequestBody(
+                recaptcha = reCaptchaParameterGenerationResult.nonEnterpriseReCaptchaToken,
+                recaptchaevent = reCaptchaParameterGenerationResult.enterpriseReCaptchaEvent,
+                username = trimmedUsername
+            ).toJson()
+        }.onFailure { e ->
+            SalesforceSDKLogger.e(TAG, "Cannot JSON encode start password reset request body due to an encoding error with message '${e.message}'.", e)
+            return UnknownError
+        }.getOrNull()
+
+        // Create the start password reset request.
+        val startPasswordResetRequest = RestRequest(
+            POST,
+            LOGIN,
+            "$loginUrl$OAUTH_ENDPOINT_HEADLESS_FORGOT_PASSWORD",
+            startPasswordResetRequestBodyString,
+            null
+        )
+
+        // Submit the start password reset request and fetch the response.
+        val startPasswordResetResponse = suspendedRestCall(startPasswordResetRequest) ?: return UnknownError
+
+        // React to the start password reset response.
+        return when (startPasswordResetResponse.isSuccess) {
+            true -> {
+                startPasswordResetResponse.consumeQuietly()
+                Success
             }
 
-            else -> reCaptchaSiteKeyId
+            else -> {
+                SalesforceSDKLogger.e(
+                    TAG,
+                    "Cannot start password reset as the Salesforce Identity API returned an unsuccessful HTTP status. '${startPasswordResetResponse.asString()}'"
+                )
+                UnknownError
+            }
         }
-        val googleCloudProjectId = when {
-            googleCloudProjectId == null -> {
-                SalesforceSDKLogger.e(TAG, "A Google Cloud project id wasn't and must be provided when using enterprise reCAPATCHA.")
-                return OtpRequestResult(nativeLoginResult = UnknownError)
+    }
+
+    override suspend fun completePasswordReset(
```

Comment:
Here's the `completePasswordReset` method to match the `startPasswordReset` method.  Everything here follows the patterns we used in 12.0 for other native login methods.